oak_functions/abi/src/lib.rs

```diff
@@ -57,6 +57,21 @@ pub struct ReportMetricRequest {
     pub value: i64,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+#[cfg(feature = "oak-tf")]
+pub enum TfModelInferError {
+    // Error when running the TensorFlow model, due to bad input tensor.
+    BadTensorFlowModelInput,
+    // Error when decoding the inference bytes.
+    ErrorDecodingInference,
+}
+
+#[derive(Serialize, Deserialize)]
+#[cfg(feature = "oak-tf")]
+pub struct TfModelInferResponse {
+    pub result: Result<Vec<u8>, TfModelInferError>,
+}
```

Comment:
Not sure why this is a struct and not just a type alias, or if we really need a newtype then a single-element tuple.

Reply:
We can only derive serialize for structs and enums. When we revisit how we serialize we can probably simplify that.